src/include/platform/DeviceControlServer.h

```diff
@@ -87,25 +88,31 @@ class DeviceControlServer final
 public:
     // ===== Members for internal use by other Device Layer components.
 
+    /**
+     * @brief
+     *  Only a single fail-safe timer is started on the device, if this function is called again
+     *  when the fail-safe timer is currently armed, the currently-running fail-safe timer will
+     *  first be cancelled, then the fail-safe timer will be re-armed.
+     */
     CHIP_ERROR ArmFailSafe(System::Clock::Timeout expiryLength);
     CHIP_ERROR DisarmFailSafe();
     CHIP_ERROR CommissioningComplete();
     CHIP_ERROR SetRegulatoryConfig(uint8_t location, const CharSpan & countryCode, uint64_t breadcrumb);
-
     CHIP_ERROR ConnectNetworkForOperational(ByteSpan networkID);
 
-    inline FabricIndex GetFabricIndex() { return mFabric; }
-    inline void SetFabricIndex(FabricIndex fabricId) { mFabric = fabricId; }
+    inline FabricIndex GetFabricIndex() { return mFailSafeContext.GetFabricIndex(); }
```

Comment:
Followup: this needs to go away.  This is not a sensible operation.

Reply:
Removed